packages/@aws-cdk/aws-ecs-patterns/lib/base/queue-processing-service-base.ts

```diff
@@ -288,6 +291,9 @@ export abstract class QueueProcessingServiceBase extends Construct {
     }
     this.cluster = props.cluster || this.getDefaultCluster(this, props.vpc);
 
+    if (props.queue && (props.retentionPeriod || props.visibilityTimeout || props.maxReceiveCount)) {
+      throw new Error('retentionPeriod, visibilityTimeout and maxReceiveCount can be set only when queue is not set. Specify them in the QueueProps of the queue');
```

Comment:
We should probably have different error messages depending on which prop is provided that shouldn't be. It may be confusing if all are listed and a customer hasn't provided them all.

Reply:
Thank you very much for your quick reviews!
Added some unit tests and fixed the error messages.